### PR TITLE
test_run_id: rewrite test in criterion

### PR DESCRIPTION
**Pull request by nobles**
In the previous implementation, all tests used the same persist file. This caused problems with criterion, because criterion - by default - executes tests in parallel. As a result the tests overwrote each others's persist file. To fix this, I needed to dedicate persist files for each tests. That's why the signature of create_persist_state() needed to be modified to accept the name of the persist file.

### CODE REVIEW

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build? (admin please type: ok to test)

**Comment by MrAnno** [comment]
@kira-syslogng ok to test

**Comment by kira-syslogng** [comment]
This user does not have permission to start the build. Can one of the admins verify this patch and start the build? (admin please type: ok to test)

**Comment by kira-syslogng** [comment]
success

File: lib/tests/test_runid.c [hunk 1]
```diff
@@ -28,24 +28,28 @@
...
+#include <criterion/criterion.h>
 
+PersistState *create_persist_state(gchar *persist_file_name);
+
+TestSuite(test_run_id, .init = app_startup, .fini = app_shutdown);
 #define RUN_ID_FIRST 1
 
 PersistState *
-create_persist_state(void)
+create_persist_state(gchar *persist_file_name)
```
Comment (Kokan) [inline lib/tests/test_runid.c]: This function is  not really needed, just calls directly the other one.

**Comment by MrAnno** [comment]
@nobles Nice. Thank you very much for the PR!

**Pull request merged**